datafusion/expr/src/udaf.rs

```diff
@@ -72,20 +76,19 @@ pub struct AggregateUDF {
 
 impl PartialEq for AggregateUDF {
     fn eq(&self, other: &Self) -> bool {
-        self.name() == other.name() && self.signature() == other.signature()
+        self.inner.equals(other.inner.as_ref()) || other.inner.equals(self.inner.as_ref())
```

Comment:
I'm not sure if we want to be so general but the issue with dynamic equality is that it might not be symmetric.

Reply:
Another possibility we could do is document that the equality test in the UDFs must be symmetric.

Reply:
The issue with `downcast_ref` is that it's one sided. Perhaps we can document that and then change the implementation of aliased UDFs to be symmetric. Should it compare the aliases as well? I'm not sure what would be correct.

Reply:
I think comparing the inner function is the most straightforward... However, you are right that it does seem like it should be comparing the aliases too probably